cluster-autoscaler/utils/kubernetes/listers.go

```diff
@@ -245,6 +246,15 @@ func (readyNodeLister *ReadyNodeLister) List() ([]*apiv1.Node, error) {
 	return readyNodes, nil
 }
 
+// Get returns the node with the given name.
+func (readyNodeLister *ReadyNodeLister) Get(name string) (*apiv1.Node, error) {
```

Comment:
Please deduplicate this.
The implementation if All and Ready node lister is almost exact. One just filters out nodes. I think we should have one implementation parametrized with filtering function. 

Can be a followup but please clean this up as you are touching this part of code.